libsodium-sys/tests/crypto/crypto_generichash.rs

```diff
@@ -95,26 +95,25 @@ fn test_crypto_generichash_multipart() {
     let m = [0u8; 64];
     let key = [0u8; crypto_generichash_KEYBYTES as usize];
 
-    let mut st = vec![0u8; unsafe { crypto_generichash_statebytes() }];
-    let pst = unsafe { mem::transmute::<*mut u8, *mut crypto_generichash_state>(st.as_mut_ptr()) };
+    let mut pst = mem::MaybeUninit::<crypto_generichash_state>::uninit();
 
     assert_eq!(
-        unsafe { crypto_generichash_init(pst, key.as_ptr(), key.len(), out.len()) },
+        unsafe { crypto_generichash_init(pst.as_mut_ptr(), key.as_ptr(), key.len(), out.len()) },
         0
     );
 
     assert_eq!(
-        unsafe { crypto_generichash_update(pst, m.as_ptr(), m.len() as u64) },
+        unsafe { crypto_generichash_update(pst.as_mut_ptr(), m.as_ptr(), m.len() as u64) },
```

Comment:
Needs an ‘assume_init()’ before this line

Reply:
Why?

Reply:
I agree, no reason to `assume_init` here.

Reply:
Memory is initialized after the call to ‘generichash_init()’ and we should strive to always be clear in our code.

Reply:
Fixed.